core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -301,7 +368,7 @@ class IndexingQueueActor(searchService: SearchService) extends Actor with ActorL
   // De-dupes files that have been updated since we were last told to
   // index them. No need to aggregate values: the latest wins. Key is
   // the URI because FileObject doesn't implement equals
-  var todo = Map.empty[String, FileObject]
+  val todo = new m.HashMap[FileObject, m.Set[FileObject]] with m.MultiMap[FileObject, FileObject]
```

Comment:
any reason you prefer a `val` of a mutable container? I think it's far better to have a `var` of an immutable structure as it makes it clear that the field is stateful.